modules/data.atmosphere/R/download.NOAA.R

```diff
@@ -0,0 +1,387 @@
+  #avaliable.  
+  if (ends_today) {
+    for (i in 1:length(cf_var_names)) {
+      gefs_date = POSIX.to.GEFSDate(end_date)
+      last_data_set_at = "0000"
+      hour = lubridate::hour(as.POSIXlt(Sys.time()))
+      continue = TRUE
+      
+      if (num_days > 0) { #The data matrix already exists (it was created in the loop above), and we just have to add to it.
+        noaa_data[[i]] = cbind(noaa_data[[i]], rnoaa::gefs(noaa_var_names[i], lat.in, lon.in, raw=TRUE, time_idx = 1, forecast_time = "0000", date=gefs_date)$data)
+      } else { #we have to make the data matrix and put it in the array.
+        #This might be the case when the start and end dates are both the current date.
+        noaa_data[[i]] = matrix(rnoaa::gefs(noaa_var_names[i], lat.in, lon.in, raw=TRUE, time_idx = 1, forecast_time = "0000", date=gefs_date)$data,
+                           nrow=21, ncol=1)
+      }
+      
+      #The "hour > 6" statement is an attempt to reconcile the current time with the GEFS data should become avaliable.  However, this
+      #is not always assured, and so the tryCatch block exists to handle cases when data are not yet avaliable.
+      if (hour > 6 && continue) {
+        noaa_data[[i]] = tryCatch({
+          last_data_set_at = "0600"
+           cbind(noaa_data[[i]], rnoaa::gefs(noaa_var_names[i], lat.in, lon.in, raw=TRUE, time_idx = 1, forecast_time = "0600", date=gefs_date)$data)
+        }, error=function(e){
+          continue=FALSE
+          last_data_set_at = "0000"
+          PEcAn.logger::logger.info(paste0("Last data avaliable for ", Sys.Date(), " is the 12:00 midnight data point (0:00)."))
+          noaa_data
+          })
+      }
+      
+      if (hour > 12 && continue) {
+        noaa_data[[i]] = tryCatch({
+          last_data_set_at = "1200"
+          cbind(noaa_data[[i]], rnoaa::gefs(noaa_var_names[i], lat.in, lon.in, raw=TRUE, time_idx = 1, forecast_time = "1200", date=gefs_date)$data)
+        }, error=function(e){
+          continue=FALSE
+          last_data_set_at = "0600"
+          PEcAn.logger::logger.info(paste0("Last data avaliable for ", Sys.Date(), " is the 6:00 a.m. data point."))
+          noaa_data
+          })
+      }
+      
+      if (hour > 18 && continue) {
+        noaa_data[[i]] = tryCatch({
+          last_data_set_at = "1800"
+          cbind(noaa_data[[i]], rnoaa::gefs(noaa_var_names[i], lat.in, lon.in, raw=TRUE, time_idx = 1, forecast_time = "1800", date=gefs_date)$data)
+        }, error=function(e){
+          continue=FALSE
+          last_data_set_at = "1200"
+          PEcAn.logger::logger.info(paste0("Last data avaliable for ", Sys.Date(), " is the 12:00 noon data point."))
+          noaa_data
+          })
+      }
+    }
+  }
+  
+  if (verbose && include_forecast) {
+    PEcAn.logger::logger.info("Downloading forecast data...")
+  }
+  
+  #This loop attaches 16 day forecast data onto the data that currently exists.
+  if (include_forecast) {
```

Comment:
still don't see the need for this separate case. Indeed, this case look the most like what we want -- I think you made everything above more complicated than it needed to be. Also, time_idx should max at 65 not 64